lib/data/locationConstraintParser.js

```diff
@@ -41,23 +41,35 @@ function parseLC() {
             clients[location].clientType = 'scality';
         }
         if (locationObj.type === 'aws_s3' || locationObj.type === 'gcp') {
-            if (process.env.CI_PROXY === 'true') {
-                locationObj.details.https = false;
-                locationObj.details.proxy = proxyAddress;
+            let connectionAgent;
+            if (config.outboundProxy.url) {
+                const options = url.parse(config.outboundProxy.url);
+                if (config.outboundProxy.certFilePaths) {
+                    const { keypath, certpath, capath } =
+                        config.outboundProxy.certFilePaths;
+                    options.cert = certpath ?
+                        fs.readFileSync(certpath, 'ascii') : undefined;
+                    options.key = keypath ?
+                        fs.readFileSync(keypath, 'ascii') : undefined;
+                    options.ca = capath ?
+                        fs.readFileSync(capath, 'ascii') : undefined;
+                    options.secureProxy = true;
+                }
+                options.keepAlive = true;
+                // whether options.secureProxy is set to true determines
+                // https or http proxy
+                connectionAgent = new HttpsProxyAgent(options);
```

Comment:
Does the construction and use of `HttpsProxyAgent` have a chance to work if `config.outboundProxy.certFilePaths` is not defined? In such case we could fall back to `https.Agent` or invalidate the config? Not sure just wondering.